docarray/array/doc_vec/doc_vec.py

```diff
@@ -116,64 +132,113 @@ def __init__(
             # from each document and put them in the corresponding column
             field_type = self.doc_type._get_field_type(field_name)
 
+            is_field_required = self.doc_type.__fields__[field_name].required
+
+            first_doc_is_none = getattr(docs[0], field_name) is None
+
+            def _check_none_field():
+
+                if is_field_required:
+                    if first_doc_is_none:
+                        raise ValueError(
+                            f'Field {field_name} is None for {docs[0]} even though it is required'
+                        )
+
+                if first_doc_is_none:
+                    for i, doc in enumerate(docs):
+                        if getattr(doc, field_name) is not None:
+                            raise ValueError(
+                                f'Field {field_name} is put to None for the first doc. This mean that all of the other docs should have this field set to None as well. This is not the case for {doc} at index {i}'
+                            )
+
+            def _check_field_not_none(field_name, doc):
```

Comment:
This helper fn and the one above have almost the same name, i think it is a bit confusing

Reply:
renamed